jupyter_rfb/_jpg.py

```diff
@@ -0,0 +1,128 @@
+import io
+import importlib
+
+
+class JpegEncoder:
+    """Base JPEG encoder class.
+
+    Subclasses must import their dependencies in their __init__,
+    and implement the _encode() method.
+    """
+
+    def encode(self, array, quality):
+        """Encode the array, returning bytes."""
+
+        quality = int(quality)
+
+        # Check types
+        if hasattr(array, "shape") and hasattr(array, "dtype"):
+            if array.dtype != "uint8":
+                raise ValueError("Image array to convert to JPEG must be uint8")
+            original_shape = shape = array.shape
+        else:
+            raise ValueError(
+                f"Invalid type for array, need ndarray-like, got {type(array)}"
+            )
+
+        # Check shape
+        if len(shape) == 2:
+            shape = shape + (1,)
+            array = array.reshape(shape)
+        if not (len(shape) == 3 and shape[2] in (1, 3)):
+            raise ValueError(f"Unexpected image shape: {original_shape}")
+
+        return self._encode(array, quality)
+
+    def _encode(self, array, quality):
+        raise NotImplementedError()
+
+
+class StubJpegEncoder(JpegEncoder):
+    """A stub encoder that returns None."""
+
+    def _encode(self, array, quality):
+        return None
+
+
+class SimpleJpegEncoder(JpegEncoder):
+    """A JPEG encoder using the simplejpeg library."""
+
+    def __init__(self):
+        self.simplejpeg = importlib.import_module("simplejpeg")
```

Comment:
Is there a benefit to using `importlib` versus just `import simplejpeg`? Seems odd to use `importlib` with a static string.

Reply:
Good point. I played a around a bit with different approaches for this module - this is mostly an artifact of that. Will change.